pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -63,17 +63,26 @@ type podNIC interface {
 	PlugPhase2(vmi *v1.VirtualMachineInstance, iface *v1.Interface, network *v1.Network, domain *api.Domain, podInterfaceName string) error
 }
 
-func getNetworksAndMultusIndices(vmi *v1.VirtualMachineInstance) (map[string]*v1.Network, map[string]int) {
+func getNetworks(vmi *v1.VirtualMachineInstance) map[string]*v1.Network {
 	networks := map[string]*v1.Network{}
-	multusIndices := map[string]int{}
 	for _, network := range vmi.Spec.Networks {
 		networks[network.Name] = network.DeepCopy()
-		if networks[network.Name].Multus != nil && !networks[network.Name].Multus.Default {
-			// multus pod interfaces start from 1
-			multusIndices[network.Name] = len(multusIndices) + 1
+	}
+	return networks
+}
+
+func getMultusIfaceNames(vmi *v1.VirtualMachineInstance) map[string]string {
```

Comment:
This does 3 things: filterNonDefaultMultusNetworks, nameSecondaryNetworksIfaces and indexIfacesNamesByNetworkName.

Reply:
Would you consider this any better?
```
func isSecondaryMultusNetwork(net *v1.Network) bool {
        return net.Multus != nil && !net.Multus.Default
}

func getSecondaryMultusIfaceName(net *v1.Network, index int) string {
        return fmt.Sprintf("net%d", multusIndex)
}       

func getMultusIfaceNames(vmi *v1.VirtualMachineInstance) map[string]string {
        multusIfaceNames := map[string]string{}
        multusIndex := 1
        for _, net := range vmi.Spec.Networks {
                if isSecondaryMultusNetwork(net) {
                        multusIfaceNames[net.Name] = getSecondaryMultusIfaceName("net%d", multusIndex)
                        multusIndex = multusIndex + 1
                } else {
                        multusIfaceNames[net.Name] = primaryPodInterfaceName
                }
        }
        return multusIfaceNames
}

```

Reply:
I think that `getMultusIfaceNames` is still doing 3 things and the abstraction is a bit odd.
I was aiming at this:
```
primaryMultusNetwork := lookupPrimaryNetwork(vmi.Spec.Networks)
secondaryMultusNetworks := filterSecondaryMultusNetworks(vmi.Spec.Networks)
multusIfacesNames := indexIfacesNamesByNetworkName(primaryMultusNetwork, secondaryMultusNetworks, generateSecondaryMultusIfaceName)
```
`generateSecondaryMultusIfaceName` can also be called from within the indexing if passing it as an argument is not of your liking.

BTW, I really do not like the `get` prefix, it is overused in most cases and makes it hard for me to understand if it is a simple getter or has a lot of logic behind it. That is why I avoid it in my suggestions.

If you have a good name for the story above, then I guess you can use it to wrap them all under one roof.
Maybe something like `createMultusIfaceByNetworkPool` ?